src/ansys/systemcoupling/core/participant/manager.py

```diff
@@ -61,10 +62,14 @@ def __init__(self, syc_session):
         self.clear()
 
     def clear(self):
+        self.__server_version = None
         self.__participants: Dict[str, ParticipantProtocol] = {}
         self.__n_connected = 0
         self.__solve_exception = None
 
+    def set_server_version(self, server_version: str) -> None:
+        self.__server_version = server_version
```

Comment:
@iboyd-ansys Passing version as an argument to add_particiant would change the function signature and affect all relevant testing, would you like me to update these testings?

Reply:
It doesn't really seem to be right to have a call to `set_server_version` just before the `add_participant` call when it is only the `add_participant` method that needs the value. However, rather than passing it into the method, another option would be to make the construction of `ParticipantManager` lazy and construct it in `Session._get_api_root`. If that were done, then the version could be passed into the constructor of `ParticipantManager`. We could then even remove the version argument from `_wrap_add_participant` and move all version checking into `add_participant`.